ossdbtoolsservice/utils/cancellation.py

```diff
@@ -10,8 +10,12 @@ class CancellationToken:
     """Token used to indicate if an operation has been canceled"""
 
     def __init__(self):
-        self.canceled = False
+        self._canceled = False
 
     def cancel(self):
         """Mark the cancellation token as canceled"""
-        self.canceled = True
+        self._canceled = True
+    
+    def hasBeenCancelled(self):
```

Comment:
define it as 

@ property
def canceled(self) -> bool:
    return self._canceled

Reply:
done